pkg/packageinstall/packageinstall.go

```diff
@@ -248,12 +237,15 @@ func (pi *PackageInstallCR) kcVersionConstraintsSatisfied(pkg *datapkgingv1alpha
 		return true
 	}
 
-	v, err := dropVersionPreReleaseAndBuild(pi.controllerVersion)
+	v, err := semver.ParseTolerant(pi.fetchFactory.NewVersionFetcher().GetKappControllerVersion())
```

Comment:
I think we should not provide a factory in this case. What we should provide is an interface that looks like this

```go
type KappControllerVersionRetriever interface {
  KappControllerVersion() (semver.Version, error)
  KubernetesVersion(saName string, specCluster *v1alpha1.AppCluster, genericOpts clusterclient.GenericOpts) (semver.Version, error)
}
```
This will allow you to inject into this struct the struct that implements this interface instead of having to build a full factory that will create the struct for you.